### PR TITLE
[#55472] Do not save access token in cookies

### PR DESCRIPTION
**Pull request by ba1ash**
https://community.openproject.org/work_packages/55472
History commits:
     https://github.com/opf/openproject/commit/ba2f5e91432321f19ac4fcaaff0898cbab9e2460
     https://github.com/opf/openproject/commit/5962623b3fd678e014233cac2344dba04aeeac5d

### CODE REVIEW

File: modules/openid_connect/lib/open_project/openid_connect/hooks/hook.rb [hunk 1]
```diff
@@ -43,20 +43,7 @@ def user_logged_in(context)
       ##
       # Once omniauth has returned with an auth hash
       # persist the access token
```
Comment (akabiru) [inline modules/openid_connect/lib/open_project/openid_connect/hooks/hook.rb]:
```suggestion
      ##
      # Called once omniauth has returned with an auth hash
      # NOTE: It's a passthrough as we no longer persist the access token into the cookie
```

File: modules/openid_connect/lib/open_project/openid_connect/hooks/hook.rb [hunk 2]
```diff
@@ -43,20 +43,7 @@ def user_logged_in(context)
...
-        access_token = auth_hash.fetch(:credentials, {})[:token]
-        # put it into a cookie
-        if controller && access_token
-          controller.send(:cookies)[:_open_project_session_access_token] = {
-            value: access_token,
-            secure: !!Rails.configuration.force_ssl
-          }
-        end
-      end
+      def omniauth_user_authorized(_context); end
```
Comment (akabiru) [inline modules/openid_connect/lib/open_project/openid_connect/hooks/hook.rb:46]: 🟢  Q: We're retaining the empty method because of the existing API?

**Review by akabiru — APPROVED** [review]
:shipit:

File: modules/openid_connect/lib/open_project/openid_connect/hooks/hook.rb (hunk 2 above)
Comment (ba1ash) [reply modules/openid_connect/lib/open_project/openid_connect/hooks/hook.rb:46]: Yes, it is very likely to be used in future.

**Comment by akabiru** [comment]
@ba1ash should this have been on `release/14.2`?

**Comment by ba1ash** [comment]
> @ba1ash should this have been on `release/14.2`?

I think it is okay to have it on dev.

**Pull request merged**